src/main/java/org/opentestsystem/shared/contentspecid/ContentSpecIdBuilder.java

```diff
@@ -0,0 +1,241 @@
+import org.opentestsystem.shared.contentspecid.exceptions.ErrorSeverity;
+import org.opentestsystem.shared.contentspecid.exceptions.IdParseException;
+import org.opentestsystem.shared.contentspecid.handlers.DomainHandler;
+import org.opentestsystem.shared.contentspecid.handlers.EmphasisHandler;
+import org.opentestsystem.shared.contentspecid.handlers.TargetHandler;
+
+import static org.apache.commons.lang3.StringUtils.defaultString;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+import static org.apache.logging.log4j.util.Strings.isBlank;
+import static org.opentestsystem.shared.contentspecid.enums.ContentSpecSubject.MATH;
+import static org.opentestsystem.shared.contentspecid.exceptions.ErrorSeverity.CORRECTED;
+import static org.opentestsystem.shared.contentspecid.exceptions.ErrorSeverity.INVALID;
+
+
+/**
+ * Builds a ContentSpecId object up from its component parts.
+ */
+public class ContentSpecIdBuilder {
+
+    private static final String NOT_APPLICABLE = "NA";
+    private static final String PRIORITY_CLUSTER = "P";
+    private static final String SUPPORTING_CLUSTER = "S";
+
+
+    private final ContentSpecSubject subject;
+    private final ContentSpecGrade grade;
+    private String claim;
+    private String target = "";
+    private String ccss = "";
+    private String domain = DomainCode.UNK.getValue();
+    private String emphasis = "";
+    private String contentCategory = "";
+    
+    private final DomainHandler domainHandler = new DomainHandler();
+    private final EmphasisHandler emphasisHandler = new EmphasisHandler();
+    private final TargetHandler targetHandler = new TargetHandler();
+
+    public static ContentSpecIdBuilder getBuilder(ContentSpecSubject subject, ContentSpecGrade grade) {
+        return new ContentSpecIdBuilder(subject, grade);
+    }
+
+    private ContentSpecIdBuilder(ContentSpecSubject subject, ContentSpecGrade grade) {
+        this.subject = subject;
+        this.grade = grade;
+    }
+
+    public ContentSpecIdBuilder claim(String claim) {
+        this.claim = defaultString(claim);
+        return this;
+    }
+
+    public ContentSpecIdBuilder target(String target) {
+        this.target = defaultString(target);
+        return this;
+    }
+
+    public ContentSpecIdBuilder ccss(String ccss) {
+        this.ccss = defaultString(ccss);
+        return this;
+    }
+
+    public ContentSpecIdBuilder domain(String domain) {
+        this.domain = defaultString(domain);
+        return this;
+    }
+
+    public ContentSpecIdBuilder emphasis(String emphasis) {
+        this.emphasis = defaultString(emphasis);
+        return this;
+    }
+
+    public ContentSpecIdBuilder contentCategory(String contentCategory) {
+        this.contentCategory = defaultString(contentCategory);
+        return this;
+    }
+
+    public ContentSpecId build() throws IdParseException {
```

Comment:
Can we document this build method either linking to the classes that validate or including the validation itself?  Just so callers know that building also validates.  Currently, users of the API would have to look through the internals of the library to understand why things don't validate.